### DISCUSSION TITLE
Can someone elaborate on the Time Intervals for the main settings and sub settings in the script files?

### DISCUSSION THREAD

**Discussion by ifmvaiea** [cyberjunky/3commas-cyber-bots#196] [General]
I'm just trying to understand better, but I can't find the answers I'm looking for in the documentation (I probably missed it). I understand that there is a Time Interval for most of the config files at the top and that is how often the script does it's next check. What I don't understand is the Time Intervals for the sub-sections. I would assume that the Time Intervals in the sub-sections must be the same as the Time Interval at the top of the config file, or shorter in duration. Despite that, if the script does another check based off of the master Time Interval at the top, what do the intervals in the sub-sections do? Will those parts of the script continue to run in the background on a second time interval before the interval at the top runs again? I just don't get it. Thanks!

**Comment by amargedon** [comment]
It's the other way around. The main time interval is for the main loop and in which all the sub sections are processed. However, it may be you have some sub sections that don't require that often processing (so they can be on a slower interval). 

So the interval of a sub section should equal the main interval, or be a multiply of that interval (or, when not a multiply it will be processed at the next main interval).

Does that give the explanation you are looking for?

**Comment by ifmvaiea** [comment]
Unfortunately it still doesn't make sense, as much as I hate to say that out loud. I think I need a practical example. Let's look at "botupdater.ini" with the example data:

```
[settings]
timezone = Europe/Amsterdam
timeinterval = 86400
debug = False
debug-log-query = False
logrotate = 14
3c-apikey = 4mzhnpio6la4h1158ylt2
3c-apisecret = 4mzhnpio6la4h1158ylt4mzhnpio6la4h1158ylt4mzhnpio6la4h1158ylt4mzhnpio6la4h1158ylt4mzhnpio6la4h1158ylt4mzhnpio6la4h1158ylt4mzhnpio6la4h1158ylt4mzhnpio6la4h1158ylt
3c-apikey-path = 
notifications = True
notify-urls = [ "tgram://9995888120:BoJPor6opeHyxx5VVZPX-BoJPor6opeHyxx5VVZPX/" ]

[bu_default]
botids = [ 123456 ]
timeinterval = 86400
allowmaxdealchange = True
allowbotstopstart = True
```

When I start botupdater.ini for the first time it turns on notifications, loads the database, fetches blacklist pairs, evaluates the details in [bu_default], decides whether to update the bots in [bu_default], then sends me a notification. Since the timeinterval is 24 hours, I'm assuming in one day it will do all of the above again.

Is that correct?

While botupdater.ini is running in the background there is a sub-section for [bu_default] bots. With the example data the same for the main and sub-section it is easy to understand that everything will update at the same time. I thought that having a sub-section interval for [bu_default] that was less than the main setting would cause the sub-section to update more often, but as I type this that wouldn't make sense because the main settings would be the same, as the main settings updates everything every 24 hours.

Is that correct?

If so, how does having a longer time interval for the sub-section work? My initial thoughts were that a longer interval for [bu_default] would be lost by the fact that the main script updates every 24 hours, wiping out the time interval for [bu_default] on it's refresh. I'm assuming that is wrong and when the main script updates it still stays open, tracking the longer interval for [bu_default] without wiping out the time tracking with the update.

Is that correct?

Do you have a practical example where someone would want to delay an update of a sub-section to a period longer than the main script updates? In my mind, if you are updating all of the data for the main script wouldn't you want to update your bots with that information as soon as it's available?

Thank you for the help and for the prompt reply! I understand everything else in your scripts.

**Reply by amargedon** [reply]
That's all correct. When you have only one bu section, it makes no sense to have different intervals.

Let's imagine you have two or more bu sections. Let's name them bu_1 and bu_2.

The first section is used for top x coins, which don't change that often. My second section is used for volatile coins, which can change rapidly.

So, I want to process the first section every 4h and the second section every 30m. So, my main interval must be 30m or less (to make sure the second section will be processed) and only every 8th processing interval the first section will be processed.

This will reduce the processing time, the number of API calls, log/debug lines and telegram notifications.

I've used these different intervals in the past especially on even shorter intervals, like 5m and 15, and working with 25+ bots.
